### PR TITLE
ci: publish all available test results on merge to master

### PR DESCRIPTION
**Pull request by danWatkinson**
## NOTE
this is deliberately raised as a PR to master. we're doing it to enable part of UCD 831 but we can (and should) have this stuff published today regardless.

## Ticket Number

UCD-1685 / UCD-831
we aren't going to cover everything in cypress tests, so we need to expose our other test-types..

## Description of change
publish all our jest tests as part of our documentation run
+ clean up the scripts a bit + move to parallel execution for extra speed and points.

## Checklist

- [ ] Requires infrastructure changes
- [ ] If adding or remove environment variables (e.g. in `docker-compose.yaml`) then I have updated the appropriate Helm chart
- [ ] I have updated the documentation accordingly
- [ ] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `master` (please only [rebase](https://github.com/foundry4/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

**Review by frank-dodion — APPROVED** [review]

File: packages/common/.gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# ignore the output of "npm run documentation"
+jest_html_reporters.html
+./coverage/*
```
Comment (MrSimonEmms) [inline packages/common/.gitignore:3]: Is this needed? We have `coverage` ignored in the root `.gitignore`

File: packages/appeals-service-api/.gitignore [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# ignore the output of "npm run documentation"
+jest_html_reporters.html
+./coverage/*
```
Comment (MrSimonEmms) [inline packages/appeals-service-api/.gitignore:3]: Is this needed? We have `coverage` ignored in the root `.gitignore`

File: packages/document-service-api/.gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# ignore the output of "npm run documentation"
+jest_html_reporters.html
+./coverage/*
```
Comment (MrSimonEmms) [inline packages/document-service-api/.gitignore:3]: Is this needed? We have `coverage` ignored in the root `.gitignore`

File: packages/forms-web-app/.gitignore [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# ignore the output of "npm run documentation"
+jest_html_reporters.html
+./coverage/*
```
Comment (MrSimonEmms) [inline packages/forms-web-app/.gitignore:3]: Is this needed? We have `coverage` ignored in the root `.gitignore`

File: packages/lpa-questionnaire-web-app/.gitignore [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# ignore the output of "npm run documentation"
+jest_html_reporters.html
+./coverage/*
```
Comment (MrSimonEmms) [inline packages/lpa-questionnaire-web-app/.gitignore:3]: Is this needed? We have `coverage` ignored in the root `.gitignore`

File: packages/lpa-questionnaire-web-app/.gitignore0 [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# ignore the output of "npm run documentation"
+jest_html_reporters.html
+./coverage/*
```
Comment (MrSimonEmms) [inline packages/lpa-questionnaire-web-app/.gitignore0:3]: Is this needed? We have `coverage` ignored in the root `.gitignore`

**Review by MrSimonEmms — CHANGES_REQUESTED** [review]
I'd suggest removing the `.gitignore` from each package (we'll forget that in future) and add `**/jest_html_reporters.html` to the root `.gitignore` instead

**Review by MrSimonEmms — APPROVED** [review]
LGTM

**Pull request merged**